### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by raphael**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -327,13 +336,14 @@ There is also a [FAQ](https://github.com/goadesign/goa/blob/v2/docs/FAQ.md) and
 a document describing
 [error handling](https://github.com/goadesign/goa/blob/v2/docs/ErrorHandling.md).
 
+If you are coming from v1 you may also want to read the 
+[Upgrading](https://github.com/goadesign/goa/blob/v2/docs/Upgrading.md) document.
+
 ### Examples
 
-The [examples](https://github.com/goadesign/goa/tree/v2/examples) directory
+The [examples](https://github.com/goadesign/goa/tree/master/examples) directory
```
Comment (arangamani) [inline README.md]: Didn't we move examples to its own repository?

File: docs/Guide.md [hunk 2]
```diff
@@ -217,50 +225,50 @@ etc.
...
-other fields are mapped to the HTTP request body.
+only one of the fields for the payload type is mapped to a HTTP header for
+example.
 
 The body attributes may also be listed explicitly using the `Body` function.
-This function accepts either a DSL listing the body attributes or the name of a
-request type attribute whose type defines the body as a whole. The latter makes
-it possible to use any arbitrary type to describe request body and not just
-object, for example, the attribute (and thus the body) could be an array.
+This function accepts either a DSL listing the corresponding payloard type
```
Comment (arangamani) [inline docs/Guide.md]: payloard -> payload

File: docs/Guide.md [hunk 3]
```diff
@@ -306,27 +324,28 @@ is dictated by the type of the attribute.
...
-`[{"name"="foo"},{"name"="bar"}]` assuming the type `Account` only has a `name`
-attribute. The same example but with the line defining the response body
-(`Body("accounts")`) removed produces response bodies of the form:
-`{"accounts":[{"name"="foo"},{"name"="bar"}]` since `accounts` isn't used to
-define headers.
+The HTTP response body for requests sent to the "index" method are of the form
+`[{"name"="foo"},{"name"="bar"}]`. The same example but with the line defining
+the response body removed (`Body("accounts")`) produces response bodies of the
+form:
+`{"accounts":[{"name"="foo"},{"name"="bar"}]` since the shape of the response
```
Comment (arangamani) [inline docs/Guide.md]: Was it meant to be `"name":"foo"` instead of `"name"="foo"`?

File: README.md (hunk 1 above)
Comment (raphael) [reply README.md]: Thanks, fixed.

File: docs/Guide.md (hunk 2 above)
Comment (raphael) [reply docs/Guide.md]: fixed

File: docs/Guide.md (hunk 3 above)
Comment (raphael) [reply docs/Guide.md]: yup, thank you! fixed.

**Pull request merged**